include/boost/geometry/algorithms/detail/disjoint/point_geometry.hpp

```diff
@@ -39,20 +39,25 @@ namespace detail { namespace disjoint
 {
 
 
-template<typename Point, typename Geometry>
-struct disjoint_point_linear
+struct disjoint_point_segment
 {
-    static inline
-    bool apply(Point const& pt, Geometry const& g)
+    template <typename Point, typename Segment>
+    static inline bool apply(Point const& point, Segment const& segment)
     {
-        return !geometry::covered_by(pt, g);
+        typedef geometry::model::referring_segment<Point const> other_segment;
+
+        other_segment other(point, point);
+        return disjoint_segment
+            <
+                Segment, other_segment
+            >::apply(segment, other);
```

Comment:
The point-segment disjointness test was simply moved from one place of the file to another. The actual implementation was not changed.

What I eliminated (but it is not visible easily in the diff) is `disjoint_point_linear`. Its implementation was the same as `reverse_covered_by`, so I simply use `reverse_covered_by` in the new version.

The changes in this file are all about code refactoring and polishing. There is nothing new.